### DISCUSSION TITLE
How to partially update the page by changing query parameters

### DISCUSSION THREAD

**Discussion by viktor32** [vercel/next.js#61739] [Help]
### Summary

Hello
I have the following task: a page includes a list of items with pagination. On the right column there are some additional widgets, at the top and bottom of the page there are banners (not global, specified for the page only).
The pagination should work with query parameters, so by clicking on the page the query parameter `page` is changed and the page is rendered. I want to use SSC as much as possible, so the list, widgets and pagination are server-side components.

The problem here is that when the query parameter is changed the entire page is received from the backend and rerendered, but actually I need to update only one block - the list of items that are paginated. 

Could you help to understand if Is there any way to implement this flow (with pagination and query changing) but keep updating only one component when the page is changed? 

Thank you

**Comment by GabenGar** [comment]
Well if you need only one block changed dynamically then it stops being a server component. At this point you need to call [`router.push()`/`router.replace()`](https://nextjs.org/docs/app/api-reference/functions/use-router#userouter) in an `onClick()` handler and depend on the page value in the `useEffect()` dependency array.
